engine/src/main/java/org/hibernate/validator/internal/util/privilegedactions/LoadClass.java

```diff
@@ -80,7 +91,9 @@ private LoadClass(String className, ClassLoader classLoader, boolean fallbackOnT
 			exception = e;
 		}
 		if ( fallbackOnTCCL ) {
-			ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+			ClassLoader contextClassLoader = initialThreadContextClassLoader != null
```

Comment:
this seems to repeat once more, maybe we can init the `contextClassLoader` in the constructor ?